### ISSUE TITLE
"Add to Slack" Button throws OAuth error

### ISSUE THREAD

**Issue by jamespsteinberg** [slackapi/bolt-python#492]
The "Add to Slack" button here https://api.slack.com/docs/slack-button fails with the error

Oops, Something Went Wrong!
Please try again from here or contact the app owner (reason: invalid_browser: This can occur due to page reload, not beginning the OAuth flow from the valid starting URL, or the /slack/install URL not using https://)

This is repeatable on Firefox & Chrome.

The https://XXX.com/slack/install button works. I can also get the "Add to Slack" button to temporarily work if I first go to the https://XXX.com/slack/install page even if I do not click anything there.

Is there a function that https://XXX.com/slack/install is calling when it loads or a setting I am missing to get this to work?

**Comment by seratch** [comment]
@jamespsteinberg 
If you are going to use a guessable value for the `state` parameter, I don't recommend the approach. The `state` parameter should be a random/unguessable value for safety. 

An alternative for your use case would be to implement your own `OAuthFlow` and to override `OAuthFlow#handle_installation(request: BoltRequest)` method to set an additional cookie value for the unique code. 

Either way, as long as you implement the `OAuthStateStore` in a proper way, the issue you've described here should not arise.

**Comment by kafejo** [comment]
I found this a bit confusing. Based on the doc recommendations I included Install button on my homepage that redirects to `{myapi}/slack/install` which again renders slack install button but I got rejected during the review that this is poor UX as a user has to click on a slack button twice. 

I had no idea how to do it only after exploring the API I noticed the `install_page_rendering_enabled` parameter on the `OAuthSettings` class. Which is `True` by default. The solution was to just disable this and then there's an immediate redirect from the `/slack/install` to the install url the endpoint generates. 

It would be nice to mention this in the Bolt oAuth2 doc.

**Comment by bhaktatejas922** [comment]
@seratch is there a way to get extra query params from the /slack/install method to the authorize url?

**Comment by bhaktatejas922** [comment]
I have no clue how to get extra query params through the install process.

**Comment by seratch** [comment]
@bhaktatejas922 As previously mentioned, you can create a custom `OAuthFlow` class and override the `OAuthFlow#handle_installation(request: BoltRequest)` method. Once you've done this, you can then pass its instance during the `App` instance initialization. This approach will allow you to access any query parameters within the `handle_installation(request)` method.

Please note, since this issue has been closed, should you have any additional questions or require further clarification, kindly open a new issue for discussion. Thank you for using bolt-python!
